py3status/modules/uptime.py

```diff
@@ -63,63 +62,26 @@ class Py3status:
     format = 'up {days} days {hours} hours {minutes} minutes'
 
     def post_config_hook(self):
-        self._decades = self.py3.format_contains(self.format, 'decades')
-        self._years = self.py3.format_contains(self.format, 'years')
-        self._weeks = self.py3.format_contains(self.format, 'weeks')
-        self._days = self.py3.format_contains(self.format, 'days')
-        self._hours = self.py3.format_contains(self.format, 'hours')
-        self._minutes = self.py3.format_contains(self.format, 'minutes')
-        self._seconds = self.py3.format_contains(self.format, 'seconds')
+        self.time_periods = {}
+        t = ['decades', 'years', 'weeks', 'days', 'hours', 'minutes', 'seconds']
+        s = [315360000, 31536000, 604800, 86400, 3600, 60, 1]
+        for unit, second in zip(t, s):
+            if self.py3.format_contains(self.format, unit):
+                self.time_periods[unit] = second
 
     def uptime(self):
-        # Units will be computed from bare seconds since timedelta only
-        # provides .days and .seconds anyway. Getting rid of the seconds
-        # part. Keeping the floating point part would make divmod return
-        # floats, and thus would require days/hours/minutes/seconds to be
-        # casted to int before formatting, which would be dirty to handle
-        # since we can't cast None to int.
         with open('/proc/uptime', 'r') as f:
             up = int(float(f.readline().split()[0]))
             offset = time() - up
 
-        cache_timeout = decades = years = weeks = days = hours = minutes = seconds = 0
-
-        # Decades
-        if self._decades:
-            decades, up = divmod(up, 315360000)  # 10 years -> decade
-            cache_timeout = 315360000
-        # Years
-        if self._years:
-            years, up = divmod(up, 31536000)  # 365 days -> year
-            cache_timeout = 31536000
-        # Weeks
-        if self._weeks:
-            weeks, up = divmod(up, 604800)  # 7 days -> week
-            cache_timeout = 604800
-        # Days
-        if self._days:
-            days, up = divmod(up, 86400)  # 24 hours -> day
-            cache_timeout = 86400
-        # Hours
-        if self._hours:
-            hours, up = divmod(up, 3600)  # 60 minutes -> hour
-            cache_timeout = 3600
-        # Minutes
-        if self._minutes:
-            minutes, up = divmod(up, 60)  # 60 seconds -> minute
-            cache_timeout = 60
-        # Seconds
-        if self._seconds:
-            seconds = up  # 1000000000 nanoseconds -> second
-            cache_timeout = 1
-
-        uptime = self.py3.safe_format(self.format, dict(
-            decades=decades, years=years, weeks=weeks, days=days,
-            hours=hours, minutes=minutes, seconds=seconds))
+        uptime = {}
+        for unit in self.time_periods:
```

Comment:
Here a specific order would be better `for x in dict` does not give a consistent order

Reply:
Good now? And do you want me to remove `tobes`? I recall you asked politely to stop doing that... and I did so politely. Now I'm politely asking you about this name.

Reply:
Either make `self.time_periods` an ordered dict or also create a `list` and use that.

as I said but 4 hours ago

>  `for x in dict` does not give a consistent order

It may in some implementation but there is no guarantee.  It would be nice if you read my comments on occasion.

Reply:
Re-check again. I pushed it to correct branch.

Reply:
@tobes Ping. First time using OrderedDict here.

Reply:
There is a good reason they are called ordered dict :)